[5033][IMP] template_content_swapper

kanda999 opened the pull request:
[5033](https://www.quartile.co/web#id=5033&menu_id=505&cids=3&action=1457&model=project.task&view_type=form)

kanda999 committed:
[IMP] template_content_swapper

yostashiro commented on template_content_swapper/models/ir_ui_view.py:
```diff
@@ -11,19 +11,17 @@
         result_str = str(result)
         lang_code = self.env.user.lang
         if values and values.get("request"):
-            # For views
```
Why remove this comment?

yostashiro commented on template_content_swapper/models/ir_ui_view.py:
```diff
@@ -11,19 +11,17 @@
         else:
             lang_match = re.search(r'data-oe-lang="([^"]+)"', result_str)
             if lang_match:
-                # For reports
```
Why remove this comment?

yostashiro commented on template_content_swapper/models/ir_ui_view.py:
```diff
@@ -11,19 +11,17 @@
-                # For reports
                 lang_code = lang_match.group(1)
-        view = self.browse(self.get_view_id(template)).sudo()
+        view = self.sudo()
```
```suggestion
```
We might as well remove this and change `view.id` below to `self.sudo().id`

AungKoKoLin1997 committed:
reflect oca changes

AungKoKoLin1997 committed:
[IMP] template_content_swapper

This PR moves the content mapping logic from _render_template to _render, which is used more consistently,
ensuring content replacement works in all rendering scenarios.

yostashiro reviewed (commented):
@AungKoKoLin1997 Please assign a proper commit message to the last commit, explaining the motivation and implications.

AungKoKoLin1997 commented:
> Please assign a proper commit message to the last commit, explaining the motivation and implications.

@yostashiro Done!

kanda999 commented:
@AungKoKoLin1997 
There are corner cases where using _render is more appropriate than _render_template. I've described the details in the ticket, so please check it and update the commit message and the code if necessary.

AungKoKoLin1997 commented:
For the CI is red, there is some issue in OCA CI. Same thing happens in oca repo.
https://github.com/OCA/manufacture/pull/1513#issuecomment-2782225750

yostashiro commented:
@AungKoKoLin1997 Please relay the change in the OCA repo.

AungKoKoLin1997 committed:
[IMP] template_content_swapper: remove sudo()

AungKoKoLin1997 commented:
I tested the login page as a public user and there is no issue without `sudo()`.
So, removing that in the latest commit.

yostashiro commented on template_content_swapper/models/ir_ui_view.py:
```diff
@@ -11,8 +11,8 @@
-    def _render_template(self, template, values=None):
-        result = super()._render_template(template, values)
+    def _render(self, values=None, engine="ir.qweb", minimal_qcontext=False):
+        result = super()._render(values, engine, minimal_qcontext)
```
To be a bit safer.
```suggestion
        result = super()._render(values, engine=engine, minimal_qcontext=minimal_qcontext)
```

AungKoKoLin1997 committed:
adj

nobuQuartile reviewed (approved):
LGTM

nobuQuartile commented:
I have finished the functional test about the view of the survey at the test env.
We can proceed with the merge after the customer has reviewed it.
